node/src/service.rs

```diff
@@ -259,35 +261,29 @@ pub fn build_check_assigned_para_id(
 fn check_assigned_para_id(
     cc_spawn_tx: UnboundedSender<CcSpawnMsg>,
     sync_keystore: KeystorePtr,
-    initial_assigned_para_id: Arc<Mutex<Option<ParaId>>>,
     client_set_aside_for_cidp: Arc<ParachainClient>,
     block_hash: H256,
 ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+    // Check current assignment
     let res = tc_consensus::first_eligible_key::<Block, ParachainClient, NimbusPair>(
         client_set_aside_for_cidp.as_ref(),
         &block_hash,
-        sync_keystore,
+        sync_keystore.clone(),
     );
     let container_chain_para_id = res.map(|x| x.1);
-    let mut initial_assigned_para_id = initial_assigned_para_id.lock().expect("poison error");
-    log::info!(
-        "ContainerChain assignment: old {:?} new {:?}",
-        initial_assigned_para_id,
-        container_chain_para_id
-    );
-    if container_chain_para_id != *initial_assigned_para_id {
-        if let Some(new_para_id) = container_chain_para_id {
-            // Spawn new container chain node
-            cc_spawn_tx.send(CcSpawnMsg::Spawn(new_para_id))?;
-        }
 
-        if let Some(prev_para_id) = *initial_assigned_para_id {
-            // Stop old container chain node
-            cc_spawn_tx.send(CcSpawnMsg::Stop(prev_para_id))?;
-        }
+    // Check assignment in the next session
+    let res2 = tc_consensus::first_eligible_key_next_session::<Block, ParachainClient, NimbusPair>(
```

Comment:
Here instead of calling two functions `first_eligible_key` and `first_eligible_key_next_session` we should call one that returns both

Reply:
yes, but we can do a refactor for this later